src/App.jsx

```diff
@@ -14,45 +17,96 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_BY_ALPHAB = 'name';
+const SORT_BY_LENGTH = 'length';
+
+function getSortedGoods(goodsArr, sortField) {
+  if (sortField !== '') {
+    goodsArr.sort((good1, good2) => {
+      switch (sortField) {
+        case SORT_BY_ALPHAB: {
+          return good1.localeCompare(good2);
+        }
+
+        case SORT_BY_LENGTH: {
+          return good1.length - good2.length;
+        }
+
+        default: {
+          return goodsArr;
+        }
+      }
+    });
+  }
+
+  return goodsArr;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
+  const isHideResetBtn = sortField || isReversed;
+
+  const resetFields = () => {
+    setSortField('');
+    setIsReversed(false);
+  };
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  const goods = getSortedGoods([...goodsFromServer], sortField);
+
+  if (isReversed) {
+    goods.reverse();
+  }
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          onClick={() => setSortField(SORT_BY_ALPHAB)}
+          className={cn(
+            'button is-info', { 'is-light': sortField !== SORT_BY_ALPHAB },
+          )}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          onClick={() => setSortField(SORT_BY_LENGTH)}
+          className={cn(
+            'button is-success', { 'is-light': sortField !== SORT_BY_LENGTH },
+          )}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          onClick={() => setIsReversed(!isReversed)}
+          className={cn(
+            'button is-warning', { 'is-light': !isReversed },
+          )}
+        >
+          Reverse
+        </button>
```

Comment:
In future, when you gonna use typescript, don't forget that you can iterate through 'enum' of sort type options, and render these buttons in easier way